DROD/EditRoomScreen.cpp

```diff
@@ -7106,6 +7106,17 @@ void CEditRoomScreen::SetSelectedObject(const UINT wObject)
 {
 	if (this->wSelectedObject != wObject)
 	{
+		// Place long monster before changing anything ...
+		if (this->eState == ES_LONGMONSTER) {
+			const bool bIsOldTileSerpent = (this->wSelectedObject == T_SERPENT || this->wSelectedObject == T_SERPENTB || this->wSelectedObject == T_SERPENTG);
```

Comment:
Do gentryii also need to be considered?

Reply:
Nope, because this case covers changing serpent type with mouse wheel/clicking the serpent again in the UI. Gentryii have only one type.

Reply:
I'll change the comments to be clearer